Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -164,14 +172,19 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
             if (!remoteAddress)
                 return cookiesSetInResponse;
 
-            auto needsThirdPartyIPAddressQuirk = [] (const URL& requestURL) {
-                // We only apply this quirk if we're already on google.com; otherwise, we would've
-                // already bailed at the top of this method, due to the request being third party.
-                // Note that this only applies to "accounts.google.com" (excluding subdomains).
-                return requestURL.host() == "accounts.google.com"_s;
+            auto needsThirdPartyIPAddressQuirk = [] (const URL& requestURL, const String& topDomainName) {
+                // We only apply this quirk if we're already on Google or youtube.com;
+                // otherwise, we would've already bailed at the top of this method, due to
+                // the request being third party.
+                auto hostName = requestURL.host();
+                if (hostName == "accounts.google.com"_s)
+                    return true;
+
+                return (topDomainName.startsWith("google."_s) || topDomainName == "youtube.com"_s)
```

Comment:
Down here, one worries that any site could add a `google.` subdomain to trigger this rule, for instance `google.some.unrelated.example.com`. However, with variable name `topRegistrableDomain` it will be clear that the rule is much more narrow than that.

Still, we need to acknowledge that there may be registrable domains starting with `google.` that don't belong to Google. You can imagine any entry on the Public Suffix List with a leading `google.`. For a soft mitigation like capped cookie lifetime, I think that's OK to keep this speedy. But it's important that harder policies don't rely on this overly relaxed rule.